src/main/java/org/folio/innreach/client/InventoryClient.java

```diff
@@ -22,4 +24,10 @@
 
   @GetMapping("/items?query=hrid=={hrId}")
   ResultList<InventoryItemDTO> getItemsByHrId(@PathVariable("hrId") String hrId);
+
+  @GetMapping("/items?query=barcode=={barcode}")
+  InventoryItemDTO getItemByBarcode(@PathVariable("barcode") String barcode);
```

Comment:
The result of query is an array (most probably with single item). It should be wrapped to `ResultList`